abstractions/python/kiota/abstractions/request_adapter.py

```diff
@@ -0,0 +1,129 @@
+from abc import ABC, abstractmethod
+from datetime import datetime
+from io import BytesIO
+from typing import List, Optional, TypeVar
+
+from .request_information import RequestInformation
+from .response_handler import ModelType, ResponseHandler
+from .serialization import Parsable, SerializationWriterFactory
+from .store import BackingStoreFactory
+
+ResponseType = TypeVar("ResponseType", str, int, float, bool, datetime, BytesIO)
+
+
+class RequestAdapter(ABC):
+    """Service responsible for translating abstract Request Info into concrete native HTTP requests.
+    """
+    # The base url for every request.
+    base_url = str
+
+    @abstractmethod
+    def get_serialization_writer_factory(self) -> SerializationWriterFactory:
+        """Gets the serialization writer factory currently in use for the HTTP core service.
+
+        Returns:
+            SerializationWriterFactory: the serialization writer factory currently in use for the
+            HTTP core service.
+        """
+        pass
+
+    @abstractmethod
+    async def send_async(
+        self, request_info: RequestInformation, type: ModelType,
```

Comment:
does python support generic constructors? will you be able to do `new type()` in the implementation?
does python support generic type constraint? Can you force modeltype to implement the parsable interface?

Reply:
Yes, generic constructors are supported. See https://github.com/microsoft/kiota/pull/1060/commits/21fe2fa4035fafded2c8921f0d46dc83ad17884c
Also generic type constraint are supported like so  `ModelType = TypeVar("ModelType", bound=Parsable)`.